src/main/java/dev/openfga/sdk/api/client/OpenFgaClient.java

```
@@ -1104,6 +1107,86 @@ public CompletableFuture<ClientListObjectsResponse> listObjects(
+                body.contextualTuples(bodyContextualTuples);
+            }
+            if (request.getContext() != null) {
+                body.context(request.getContext());
+            }
+        }
+
+        if (options != null) {
+            if (options.getConsistency() != null) {
+                body.consistency(options.getConsistency());
+            }
+
+            // Set authorizationModelId from options if available; otherwise, use the default from configuration
+            String authorizationModelId = !isNullOrWhitespace(options.getAuthorizationModelId())
+                    ? options.getAuthorizationModelId()
+                    : configuration.getAuthorizationModelId();
+            body.authorizationModelId(authorizationModelId);
+        } else {
+            body.setAuthorizationModelId(configuration.getAuthorizationModelId());
+        }
+
+        Configuration config = configuration.override(new ConfigurationOverride().addHeaders(options));
+
+        return call(() -> streamedListObjectsApi.streamedListObjects(storeId, body, config))
+                .thenApply(response -> {
+                    StreamingResponseBody srb = response.getData();
+                    java.io.BufferedReader reader = new java.io.BufferedReader(
+                            new java.io.InputStreamReader(srb.getBody(), java.nio.charset.StandardCharsets.UTF_8));
+                    StreamedResponseIterator iterator =
+                            new StreamedResponseIterator(reader, apiClient.getObjectMapper());
+                    Stream<StreamedListObjectsResponse> stream = java.util.stream.StreamSupport.stream(
+                            ((Iterable<StreamedListObjectsResponse>) () -> iterator).spliterator(), false);
+                    return stream.onClose(() -> {
+                        try {
+                            iterator.close();
+                        } catch (java.io.IOException ignore) {
```

should we be ignoring exceptions closing the iterator or response body? Seems like it could cause issues to swallow these errors. If we can recover we should, if not can we throw back to caller with applicable information? Or if there is good reason to swallow exceptions, let's add clear code comment explaining why.